### DISCUSSION TITLE
Device - for Server specs

### DISCUSSION THREAD

**Discussion by JHarrison712** [netbox-community/netbox#12899] [Ideas]
In Netbox shouldn't Devices have a field for resource such has the following information

CPU, Memory, OS Hard Drive, Shared Storage ( Maybe that attach to another device ) 

Because when you go down to the Virtualization section, Under Virtual Machine you have resources
vCPU, Memory, Disk, and that is linked to the cluster / Device but in the device or cluster it doesn't have processor or cpu or etc? Not sure if I'm missing a step?

**Comment by candlerb** [comment]
For Devices, which have real physical hardware, you add resources as "Inventory Items" if you want to track them - this lets you track down to the level of individual DIMMs, socketed CPUs, power supplies etc.

If you still want a summary field such as "number of CPUs" or "total amount of RAM" then you can add [Custom Fields](https://docs.netbox.dev/en/stable/customization/custom-fields/) to Device for this purpose.

These fields are available by default on VMs because they are commonly needed at the time of provisioning a VM.

**Comment by bitcollector1** [comment]
Here is an example of how we use the inventory 
<img width="1315" alt="Screenshot 2023-06-14 at 12 16 18" src="https://github.com/netbox-community/netbox/assets/50723251/d9a0c939-dc8c-49e9-a543-2f61d87f0bd9">

<img width="1309" alt="Screenshot 2023-06-14 at 12 19 19" src="https://github.com/netbox-community/netbox/assets/50723251/167a910c-4b9a-4a7b-9b51-6662e3f25aea">

**Comment by JHarrison712** [comment]
Hmm, I like your design but I feel like it should be implemented as the Virtualization looks back into the device on where its hosted as. But I do understand I guess what I can do is put it under the comments sections of the device build

**Reply by molinarif02** [reply]
I completely agree, "Custom Fields" should not be the way to do this.
"Inventory Items" are too complicated to maintain and don't have the kind of data fields we need, which would lead to even more "Custom Fields".
I don't think it would be difficult to implement either...
